import-export-cli/integration/apiProduct_test.go

```diff
@@ -0,0 +1,774 @@
+	publishAPI(dev, apiPublisher, apiPublisherPassword, dependentAPI2ofEnv1.ID)
+	publishAPI(prod, apiPublisher, apiPublisherPassword, dependentAPI2ofEnv2.ID)
+
+	// Map the real name of the APIs with the APIs of env1
+	apisList := map[string]*apim.API{
+		"PizzaShackAPI":   dependentAPI1ofEnv1,
+		"SwaggerPetstore": dependentAPI2ofEnv1,
+	}
+
+	// Add the API Product to env1
+	apiProduct := addAPIProductFromJSON(t, dev, apiPublisher, apiPublisherPassword, apisList)
+
+	args := &apiProductImportExportTestArgs{
+		apiProductProvider:   credentials{username: apiPublisher, password: apiPublisherPassword},
+		ctlUser:              credentials{username: adminUsername, password: adminPassword},
+		apiProduct:           apiProduct,
+		srcAPIM:              dev,
+		destAPIM:             prod,
+		importApisFlag:       false,
+		updateApisFlag:       true,
+		updateApiProductFlag: false,
+	}
+
+	// Import the API Product without importing the dependent APIs to env2 but updating those
+	validateAPIProductExportImportPreserveProvider(t, args)
+}
+
+// Export an API Product with its dependent APIs from one environment and import to another environment freshly as super tenant admin
+// and try to update that API Product (without updating dependent APIs)
+func TestExportImportApiProductAdminSuperTenantUserWithUpdateApiProduct(t *testing.T) {
+	adminUsername := superAdminUser
+	adminPassword := superAdminPassword
+
+	apiCreator := creator.UserName
+	apiCreatorPassword := creator.Password
+
+	apiPublisher := publisher.UserName
+	apiPublisherPassword := publisher.Password
+
+	dev := apimClients[0]
+	prod := apimClients[1]
+
+	// Add the first dependent API to env1
+	dependentAPI1 := addAPI(t, dev, apiCreator, apiCreatorPassword)
+	publishAPI(dev, apiPublisher, apiPublisherPassword, dependentAPI1.ID)
+
+	// Add the second dependent API to env1
+	dependentAPI2 := addAPIFromOpenAPIDefinition(t, dev, apiCreator, apiCreatorPassword)
+	publishAPI(dev, apiPublisher, apiPublisherPassword, dependentAPI2.ID)
+
+	// Map the real name of the API with the API
+	apisList := map[string]*apim.API{
+		"PizzaShackAPI":   dependentAPI1,
+		"SwaggerPetstore": dependentAPI2,
+	}
+
+	// Add the API Product to env1
+	apiProduct := addAPIProductFromJSON(t, dev, apiPublisher, apiPublisherPassword, apisList)
+
+	args := &apiProductImportExportTestArgs{
+		apiProductProvider:   credentials{username: apiPublisher, password: apiPublisherPassword},
+		ctlUser:              credentials{username: adminUsername, password: adminPassword},
+		apiProduct:           apiProduct,
+		srcAPIM:              dev,
+		destAPIM:             prod,
+		importApisFlag:       true,
+		updateApisFlag:       false,
+		updateApiProductFlag: false,
+	}
+
+	t.Cleanup(func() {
+		apiList := getAPIs(prod, apiCreator, apiCreatorPassword)
+		for _, api := range apiList.List {
+			deleteAPI(t, prod, api.ID, apiCreator, apiCreatorPassword)
+		}
+	})
```

Comment:
Any reason why we are calling t.Cleanup within the test body? This should be moved into the helper validation function such as _validateAPIProductExportImportPreserveProvider()_ to remove unwanted clutter from the actual tests

Reply:
If we move this to _validateAPIProductExportImportPreserveProvider_ function then a problem may occur in line 164.

Line 164 belongs to the _TestExportImportApiProductAdminSuperTenantUserWithoutImportApis_ test case. That test case will add 2 APIs to each env at the beginning.
When adding those APIs, internally *apim.AddAPI will be called twice for each env. That means when cleaning, it will call the DeleteAPI function twice for each env as well. So all the APIs will be deleted. 
So if I move this code segment to validateAPIProductExportImportPreserveProvider, the APIs will be tried to get deleted again, which will result in an error.

Reply:
Ok, lets try and handle the deletion inside the actual method that is being used to add the API. Point is lets try minimize boiler plate code in the test case function.

Reply:
Sure, will do the changes